src/rpc/methods/state.rs

```diff
@@ -1176,3 +1185,39 @@ pub async fn state_deal_provider_collateral_bounds<DB: Blockstore + Send + Sync
         min: TokenAmount::from_atto(min),
     })
 }
+
+pub enum StateGetBeaconEntry {}
+
+impl RpcMethod<1> for StateGetBeaconEntry {
+    const NAME: &'static str = "Filecoin.StateGetBeaconEntry";
+    const PARAM_NAMES: [&'static str; 1] = ["epoch"];
+    const API_VERSION: ApiVersion = ApiVersion::V1;
+
+    type Params = (LotusJson<ChainEpoch>,);
+    type Ok = LotusJson<BeaconEntry>;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore>,
+        (LotusJson(epoch),): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        {
+            let genesis_timestamp = ctx.chain_store.genesis_block_header().timestamp as i64;
+            let block_delay = ctx.state_manager.chain_config().block_delay_secs as i64;
+            // Give it a 1s clock drift buffer
+            let epoch_timestamp = genesis_timestamp + block_delay * epoch + 1;
+            let now_timestamp = chrono::Utc::now().timestamp();
+            match epoch_timestamp.saturating_sub(now_timestamp) {
+                diff if diff > 0 => {
+                    tokio::time::sleep(Duration::from_secs(diff as u64)).await;
+                }
+                _ => {}
+            };
```

Comment:
Might be simpler to just sleep for zero in case it's negative. Not a huge deal obviously.
```suggestion
            let diff = epoch_timestamp.saturating_sub(now_timestamp).max(0) as u64;
            tokio::time::sleep(Duration::from_secs(diff)).await;
```